EFCache/CacheTransactionHandler.cs

```diff
@@ -50,14 +50,27 @@ public virtual void InvalidateSets(DbTransaction transaction, IEnumerable<string
         {
             if (transaction == null)
             {
-                ResolveCache(connection).InvalidateSets(entitySets);
+                var cache = ResolveCache(connection);
+                cache.InvalidateSets(entitySets);
             }
             else
             {
                 AddAffectedEntitySets(transaction, entitySets);
             }
         }
 
+        public object Lock(IEnumerable<string> entitySets)
+        {
+            if (!(_cache is ILockableCache lockableCache)) return null;
+			return lockableCache.Lock(entitySets);
+		}

```

Comment:
@jkdufair there appear to be mixed space and tab whitespace in some places

Reply:
Yeah I accidentally used tabs in some places. Once this is approved, I'll convert it all back to spaces.